app/src/main/java/com/asfoundation/wallet/verification/intro/VerificationIntroFragment.kt

```diff
@@ -0,0 +1,329 @@
+  private lateinit var adyenSecurityCodeLayout: TextInputLayout
+  private lateinit var adyenCardImageLayout: RoundCornerImageView
+  private lateinit var adyenSaveDetailsSwitch: SwitchCompat
+
+  private var isStored = false
+  private var paymentDataSubject: ReplaySubject<AdyenCardWrapper>? = null
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+
+    require(context is VerificationActivityView) {
+      throw IllegalStateException(
+          "Wallet Verification Intro must be attached to Wallet Verification Activity")
+    }
+    activityView = context
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    paymentDataSubject = ReplaySubject.createWithSize(1)
+  }
+
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                            savedInstanceState: Bundle?): View? {
+    return inflater.inflate(R.layout.fragment_verification_intro, container, false)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    setupUi()
+    presenter.present(savedInstanceState)
+  }
+
+  override fun onSaveInstanceState(outState: Bundle) {
+    super.onSaveInstanceState(outState)
+    if (this::adyenCardNumberLayout.isInitialized) {
+      outState.apply {
+        putString(CARD_NUMBER_KEY, adyenCardNumberLayout.editText?.text.toString())
+        putString(EXPIRY_DATE_KEY, adyenExpiryDateLayout.editText?.text.toString())
+        putString(CVV_KEY, adyenSecurityCodeLayout.editText?.text.toString())
+        putBoolean(SAVE_DETAILS_KEY, adyenSaveDetailsSwitch.isChecked)
+      }
+    }
+  }
+
+  private fun setupUi() {
+    setupAdyenLayouts()
+    setupCardConfiguration()
+  }
+
+  private fun setupAdyenLayouts() {
+    adyenCardNumberLayout = adyen_card_form.findViewById(R.id.textInputLayout_cardNumber)
+    adyenExpiryDateLayout = adyen_card_form.findViewById(R.id.textInputLayout_expiryDate)
+    adyenSecurityCodeLayout = adyen_card_form.findViewById(R.id.textInputLayout_securityCode)
+    adyenCardImageLayout = adyen_card_form.findViewById(R.id.cardBrandLogo_imageView)
+    adyenSaveDetailsSwitch = adyen_card_form.findViewById(R.id.switch_storePaymentMethod)
+
+    adyenCardNumberLayout.editText?.imeOptions = EditorInfo.IME_FLAG_NO_EXTRACT_UI
+    adyenExpiryDateLayout.editText?.imeOptions = EditorInfo.IME_FLAG_NO_EXTRACT_UI
+    adyenSecurityCodeLayout.editText?.imeOptions = EditorInfo.IME_FLAG_NO_EXTRACT_UI
+
+    adyenSaveDetailsSwitch.run {
+
+      val params: LinearLayout.LayoutParams = this.layoutParams as LinearLayout.LayoutParams
+      params.topMargin = 2
+
+      layoutParams = params
+      isChecked = true
+      textSize = 14f
+      text = getString(R.string.dialog_credit_card_remember)
+    }
+
+    val height = resources.getDimensionPixelSize(R.dimen.adyen_text_input_layout_height)
+
+    adyenCardNumberLayout.minimumHeight = height
```

Comment:
Add adyenCardNumberLayout.errorIconDrawable = null here
Note for the future: We have this code repeated in now 3 places, we should try to move this to a custom view maybe?

Reply:
If this is common yeah we should probably do this at some point